common/static/common/js/discussion/discussion_router.js

```diff
@@ -127,17 +135,14 @@
             };
 
             DiscussionRouter.prototype.navigateToThread = function(thread_id) {
-                var thread;
+                var thread, targetThreadRoute;
                 thread = this.discussion.get(thread_id);
-                return this.navigate("" + (thread.get("commentable_id")) + "/threads/" + thread_id, {
-                    trigger: true
-                });
+                targetThreadRoute = getSingleThreadRoute(thread.get("commentable_id"), thread_id);
+                this.navigate(targetThreadRoute, { trigger: true});
```

Comment:
(FYI our new JS styleguide says not to add extra space inside single-line objects, e.g., `{trigger: true}` - but we're not enforcing the new style in edx-platform yet, so for this PR either spaces to either side or no spaces is fine, the eslint autofixer will switch them all to no-spaces later)